File: python-project-euler-100/p004.py
from Solution import Solution

class P004(Solution):
    def is_palindromic(self, number):
        number_as_string = str(number)
        reverse = number_as_string[::-1]
        if reverse == number_as_string:
            return True
        else:
            return False

    def solve(self):
        self.problem_number = 4
        start = 99
        finish = 999
        result = 0
        not_done = True
        i = finish
        while not_done and (i > start):
            for j in range(finish, start, -1):
                number = i * j
                if self.is_palindromic(number) and (number > result):
                    result = number
                    break
            i -= 1
        return result

def main():
	P004().run()

if __name__ == "__main__":
	main()

File: java-project-euler-100/src/main/java/net/talaatharb/projecteuler100/P004.java
package net.talaatharb.projecteuler100;

/**
 * @author mharb
 *
 */
public class P004 extends Solution {

	/**
	 * Problem solution entry point
	 * 
	 * @param args
	 */
	public static void main(String[] args) {
		new P004().run();
	}

	private boolean isPalindromic(int number) {
		final String numberAsString = String.valueOf(number);
		final int length = numberAsString.length();
		boolean result = true;
		int j = 0;

		for (int i = 0; i < length / 2; i++) {
			j = length - i - 1;
			if (numberAsString.charAt(i) != numberAsString.charAt(j)) {
				result = false;
				break;
			}
		}
		return result;
	}

	@Override
	public Number solve() {
		problemNumber = 4;
		int start = 99;
		int finish = 999;
		int result = 0;
		int i = finish;

		while (i > start) {

			for (int j = finish; j > start; j--) {

				int number = i * j;
				if (number > result && isPalindromic(number)) {
					result = number;
					break;
				}
			}
			i--;
		}

		return result;
	}
}
